packages/hooks/src/hooks/usePoller.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useRef } from "react";
+
+const usePoller = (func: () => void, delay: number): void => {
```

Comment:
Are we sure that the functions we pass to this hook will always return `void`? Most web3 contract calls return a `Promise`

Reply:
I typically wrap any code I use with this hook into a function that returns void. You could make it super generic and make it of type any if you wanted to.